src/Microsoft.DotNet.Build.Tasks/ParsePackageNames.cs

```diff
@@ -31,7 +31,7 @@ public override bool Execute()
                     Log.LogWarning("PackageDrop does not exist - '{0}'", packageDrop);
                     continue;
                 }
-                IEnumerable<ITaskItem> packages = Directory.GetFiles(packageDrop)?.Select(f => new TaskItem(f));
+                IEnumerable<ITaskItem> packages = Directory.GetFiles(packageDrop)?.Select(f => new TaskItem(Path.GetFileNameWithoutExtension(f)));
```

Comment:
coding feedback. You should try to avoid using "?." in these type of contexts. I don't think that return value can ever be null and if it were you would just be propagating your error further by now having a packages that is null and would nullref later. While it seems like a good idea to use everywhere I think you really need to think about when you use "?.", and the implications of the different code paths if the property is null.